statsforecast/models.py

```diff
@@ -17,55 +19,141 @@
 from .arima import auto_arima_f, forecast_arima, fitted_arima
 from .ets import ets_f, forecast_ets
 
-# Cell
-def auto_arima(X: np.ndarray, # time series
-               h: int, # forecasting horizon
-               future_xreg=None, #future regressors
-               fitted: bool=False, # fitted values
-               season_length: int=1, # season_length
-               approximation: bool=False, # approximation
-               level: Optional[Tuple[int]] = None # level
-    ) -> np.ndarray:
-    y = X[:, 0] if X.ndim == 2 else X
-    xreg = X[:, 1:] if (X.ndim == 2 and X.shape[1] > 1) else None
-    with np.errstate(invalid='ignore'):
-        mod = auto_arima_f(
-            y,
-            xreg=xreg,
-            period=season_length,
-            approximation=approximation,
-            allowmean=False, allowdrift=False #not implemented yet
-        )
-    fcst = forecast_arima(mod, h, xreg=future_xreg, level=level)
-    mean = fcst['mean']
-    if fitted:
-        return {'mean': mean, 'fitted': fitted_arima(mod)}
-    if level is None:
+# %% ../nbs/models.ipynb 6
+class _TS:
+    
+    def new(self):
+        b = type(self).__new__(type(self))
+        b.__dict__.update(self.__dict__)
+        return b
+
+# %% ../nbs/models.ipynb 8
+class AutoARIMA(_TS):
+    
+    def __init__(
+            self, 
+            season_length: int = 1, # Number of observations per cycle
```

Comment:
Are we going to slowly add options to these new classes to leverage all the options from `auto_arima_f`?

Also we are missing documentation in the constructor.
If the main interaction of high level API will be this classes users will appreciate documentation in the class first before the documentation of the function/